msteams-platform/graph-api/proactive-bots-&-messages/graph-proactive-bots-&-messages.md

```diff
@@ -0,0 +1,79 @@
+
+## What is proactive app installation in Teams?
+
+Before your bot can proactively message a user, it needs to be installed either as a personal app, or in a team where the user is a member. At times,  you may need to proactively message users that have _not_ installed or previously interacted with your app. For instance, the need to message vital information to everyone in your organization. For that scenario you can use the Microsoft Graph API to proactively install your bot for your users and cache the necessary values received by your bot at install.
+
+Microsoft Graph [teamsAppInstallationresource type](/graph/api/resources/teamsappinstallation?view=graph-rest-1.0) permissions allow your bot to manage its installation lifecycle for all user (personal ) or team (group) scopes within the Microsoft Teams platform:
+
+|Application permission | Description|
+|------------------|---------------------|
+|`TeamsAppInstallation.ReadWriteSelfForUser.All`|Allows a Teams app to read, install, upgrade, and uninstall itself to any **user**, without a prior sign in or use.|
+|`TeamsAppInstallation.ReadWriteSelfForTeam.All`|Allows a Teams app to read, install, upgrade, and uninstall itself in any **team**, without a prior sign in or use.|
+
+>[!NOTE]
+>
+> * Your bot must have _application permissions_ not _user delegated permissions_ because the installation is not for yourself but for others.
+>
+> * An Azure AD tenant administrator must [explicitly grant permissions to an application](/graph/security-authorization#grant-permissions-to-an-application). After an application is granted permissions, _all_ members of the Azure AD tenant will gain the granted permissions.
+
+## Enable proactive app installation and messaging
+
+ > [!IMPORTANT]
+> Microsoft Graph will only install apps published in your organization's [app catalog](../../concepts/deploy-and-publish/overview.md#publish-to-your-organizations-app-catalog) or [AppSource](https://appsource.microsoft.com/).
+
+### ✔ Determine whether your bot is currently installed for all intended users
+
+**HTTP GET** request:
+
+```http
+GET /users/{user-id}/teamwork/installedApps?$expand=teamsAppDefinition&$filter=teamsAppDefinition/teamsAppId eq '{teamsAppid}'
+```
+
+### ✔ Install your app for subsequent users
+
+**HTTP POST** request:
+
+```http
+POST /users/{user-id}/teamwork/installedApps
+{
+   "teamsApp@odata.bind" : "https://graph.microsoft.com/beta/appCatalogs/teamsApps/{teamsAppid}"
+}
+```
+
+### ✔ Retrieve conversation **chatId**
+
+**HTTP GET** request:
+
+```http
+ GET https://graph.microsoft.com/beta/users/{user-id}/teamwork/installedApps/{installation-id}/chat
+```
+
+### ✔ Send proactive messages
+
+Once your bot has been added to for a user or team and has acquired the necessary user  information, it can [Send proactive messsages](/azure/bot-service/bot-builder-howto-proactive-message?view=azure-bot-service-4.0&tabs=csharp)
```

Comment:
The linked page goes on for quite a while without ever quite telling you what code you need to write. This page is more useful:
https://docs.microsoft.com/en-us/microsoftteams/platform/bots/how-to/conversations/send-proactive-messages?tabs=dotnet

Even that, though, takes a while to get to the punchline -- feels like we need in-line the code into this page:
// Construct the message to post to conversation
Activity newActivity = new Activity()
{
    Text = "Hello",
    Type = ActivityTypes.Message,
    Conversation = new ConversationAccount
    {
        Id = response.Id  // from GET https://graph.microsoft.com/beta/users/{user-id}/teamwork/installedApps/{installation-id}/chat
    },
};

// Post the message to chat conversation with user
await client.Conversations.SendToConversationAsync(newActivity, response.Id);

(And no, I've not tested that code -- probably best to do so before publishing this :) )

Reply:
I think it's important, though, to make this process as easy as it can be to learn -- we get a fair amount of feedback that people find proactive messaging/installation too hard, and a lot of this is the conceptual hurdle of "bots are hard". Telling them that they just need a bot with five lines of code in it can make a big difference. (And yeah, I will also admit that calling four different APIs when you really wanted one API is harder than it needs to be, but that's the API we have today)